components/function-controller/internal/controllers/serverless/function_reconcile.go

```diff
@@ -81,6 +83,10 @@ func (r *FunctionReconciler) Reconcile(request ctrl.Request) (ctrl.Result, error
 	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
Please, consider adding  `context.WithTimeout` instead of `WithCancel`. It will prevent situations when the Controller is  overflowed because it received too many requests that he can't finish